src/rosdistro/distribution_cache_generator.py

```diff
@@ -193,7 +193,11 @@ def _get_cached_distribution(index, dist_name, preclean=False, ignore_local=Fals
         # since format 2 of the index file might contain a single value rather then a list
         if not isinstance(rel_file_data, list):
             rel_file_data = [rel_file_data]
-        # update cache with current distribution file
+        # if we're not including the source portion of the cache, strip it out of the existing cache
+        # in order to skip the potentially lengthy cache invalidation process.
+        if not include_source:
+            cache.source_repo_package_xmls = {}
+        # update cache with current distribution file, which filters existing cache by validity.
```

Comment:
I agree with you statement. The ROS build farm will likely not enable `--include-source` until we have a `--retain-source-cache` option.